### DISCUSSION TITLE
Support for Actix

### DISCUSSION THREAD

**Discussion by nonnish** [serenity-rs/serenity#1173] [Ideas]
This would be pretty convenient for things like running a Discord bot and a web server together. So making a GET request to some location in the web server triggers an event in the Discord bot.

After looking at the Serenity example code (basic ping-pong bot), I thought that the best way would be to make Client implement Actor. "Handler" can implement Actor too for the user to implement their own handling logic, and any Actix Message sent to the Client will be propagated to the Handler.

I'm sure there's a better way to do this though, what do you think?

**Comment by iddm** [comment]
I suppose anyone can do the same using a wrapper `struct` objects for which you implement all the needed traits. I can't also see the default behaviour you would like to have and guarantees that no one else would want another. Anyway, it sounds like you are going to create a project on the level higher than the library. This library is about discord and a framework which can be used to create a discord bot. What you are proposing isn't something that can be considered a part of a framework but already something that can be considered usage of the framework, the implementation. We can't add into the Qt framework anything we want, because other users may not want to have it at all or in a different way or just the goals of the framework aren't aligned with what we want to add. What the framework should be is to be easy to use and allow you to reach the goal this framework is created to help achieve - creating a discord bot. Adding features like you are describing isn't related to a discord bot creation.
This is just my own opinion though, I am not the one who can make decisions here, so I'd also like to receive some feedback on it.

**Reply by nonnish** [reply]
You are completely right. I think that making a separate project that depends on this one would be much more feasible. Like "actix-serenity" or something like that. However creating a new project entirely and maintaining it seems like overdoing it when the wrapper method exists. Not saying that it's a bad idea, someone certainly could pick up on it.

I have attempted to think about how to do the wrapper method you described earlier though. One thing stopping me from actually doing it is getting a context handle. For example, say I implement Actor and then implement a handler for a message. How should I go about getting a mutable reference to Context in the message handler?
